vision/imageloader/imageloader.go

```diff
@@ -224,3 +229,22 @@ func decodeImage(buffer []byte, colorSpace string) (gocv.Mat, error) {
 	}
 	return m, e
 }
+
+func init() {
+	// We use these background threads to call `gocv`. This is because `gocv` makes
+	// `Cgo` calls extensively, if we call `gocv` directly in goroutines(each epoch
+	// creates a new goroutine), the `Cgo` calls will cause Go runtime to create too
+	// many threads.
+	go func() {
+		runtime.LockOSThread()
```

Comment:
Calling `runtime.LockOSThread() ` would cause `Nan` loss at the second epoch, c.f. #358 , does this PR has the same  issue?

Reply:
No. This PR doesn't have the problem.